src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ILEmit/ILEmitResolverBuilderContext.cs

```diff
@@ -9,8 +9,8 @@ namespace Microsoft.Extensions.DependencyInjection.ServiceLookup
 {
     internal sealed class ILEmitResolverBuilderContext
     {
-        public ILGenerator Generator { get; set; }
-        public List<object> Constants { get; set; }
-        public List<Func<IServiceProvider, object>> Factories { get; set; }
+        public ILGenerator Generator { get; set; } = null!;
```

Comment:
Can we instead add a `ctor` that takes the Generator? We should be able to safely do this since the type is internal.

Reply:
Wasn't sure if it's ok to modify internal apis in nullable PR. I'll add the constructor